q2_moshpit/kraken2/tests/test_database.py

```diff
@@ -307,6 +327,39 @@ def test_fetch_db_collection_success(
         mock_find.assert_called_once_with("viral", ANY)
         mock_tqdm.assert_not_called()
 
+    @patch("requests.get")
+    @patch("tarfile.open")
+    @patch(
+        "q2_moshpit.kraken2.database._find_latest_db",
+        return_value="kraken/tmp/16S_Greengenes13.5.tgz"
+    )
+    @patch("q2_moshpit.kraken2.database.tqdm")
+    def test_fetch_db_collection_16S_success(
+            self, mock_tqdm, mock_find, mock_tarfile_open, mock_requests_get
+    ):
+        mock_requests_get.side_effect = [
+            MagicMock(status_code=200),
+            MagicMock(
+                status_code=200,
+                iter_content=lambda chunk_size: self.tar_chunks,
+                headers={}
+            )
+        ]
+        mock_tarfile_open.return_value.__enter__.return_value = MagicMock()
+
+        _fetch_db_collection("greengenes", "/tmp")
+
+        mock_requests_get.has_calls([
+            call(S3_COLLECTIONS_URL),
+            call(f"{S3_COLLECTIONS_URL}/kraken/16S_Greengenes13.5.tgz",
+                 stream=True)
+        ])
+        mock_tarfile_open.assert_called_once_with(
+            "/tmp/16S_Greengenes13.5.tgz", "r:gz"
+        )
+        mock_find.assert_called_once_with("greengenes", ANY)
+        mock_tqdm.assert_not_called()
```

Comment:
I think we should also have some assertions that will check your new `if` statement inside the `_fetch_db_collection` function. Or perhaps we could even have some mock files and test this part without mocking any function calls - you can try and see what's easier.

Reply:
I completely rewrote the tests for the 16S sequences and with this, I think the if block is tested for all 4 16S database types, however since I am still a beginner in testing you can let me know if the new tests do not do what you suggest here.😊

Reply:
Now we have tests for the method as well, as we discussed on Monday.